### PR TITLE
HARP-13055: Add individual TextElementStates for line markers

### PR DESCRIPTION
**Pull request by StefanDachwitz**
Signed-off-by: stefan.dachwitz <stefan.dachwitz@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/harp-mapview/lib/text/TextElementState.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export class TextElementState {
      * for text part and icon, there is no separate array of [[RenderState]]s for the text parts
      * of the line markers.
      */
-    private m_iconRenderStates?: RenderState | RenderState[];
+    private m_iconRenderState?: RenderState;
```
Comment (atomicsulfate) [inline @here/harp-mapview/lib/text/TextElementState.ts:38]: Doc needs update.

File: @here/harp-mapview/lib/text/TextElementState.ts [hunk 2]
```diff
@@ -248,72 +235,89 @@ export class TextElementState {
...
+    }
+}
+
+/**
+ * Test if a {@link TextElementState} is of type {@link LineMarkerElementState}
+ * @param state - Text element state to test.
+ */
+export function isLineMarkerElementState(state: TextElementState) {
+    return (state as any).m_lineMarkerIndex !== undefined;
+}
```
Comment (atomicsulfate) [inline @here/harp-mapview/lib/text/TextElementState.ts:323]: To make it simpler, we can use TextElementState class for line markers as well and just pass and optional index to the constructor.
Then change the position getter to use the index if defined.

**Review by atomicsulfate — CHANGES_REQUESTED** [review]
Have a look at the simplification I suggest.
Better also double check any performance impact by running the macbook perf tests.

**Comment by StefanDachwitz** [comment]
macbook tests please

**Review by atomicsulfate — APPROVED** [review]

**Pull request merged**